### PR TITLE
Added MFC-9330CDW

### PR DESCRIPTION
**Pull request by Pistos**
Based on ebuild for MFC-9320CW.

### CODE REVIEW

**Comment by stefan-langenmaier** [comment]
I don't have the printer but looks good to me.

Thanks a lot for the pull request!

**Comment by Pistos** [comment]
Thanks for merging!

**Comment by Pistos** [comment]
```Brother_MFC-9330CDW: File "/usr/libexec/cups/filter/brother_lpdwrapper_mfc9330cdw" not available: No such file or directory```

If I symlink `brother_lpdwrapper_mfc9330cdw -> ../../../../opt/brother/Printers/mfc9330cdw/lpd/filtermfc9330cdw` under the `/usr/libexec/cups/filter` dir, then it seems to get things going.  (Though, at the moment, I have yet to print a single page yet.)

If you had any suggestions, I'd be interested to hear them.

**Comment by stefan-langenmaier** [comment]
When do you have that problem? During the install phase of the ebuild or during the merge phase?

Can you give me some more informations how you produced this error message.

How did you create the symlink manually?

**Comment by Pistos** [comment]
Sorry, that error came from the CUPS error logs.  I should have been specific.

I symlinked it with `ln -s` from the above-mentioned dir.

I tried to find where `brother_lpdwrapper_mfc9330cdw` was defined so that I might correct any problem with the ebuild I submitted, but I couldn't find it.

Basically, there was an existing symlink which I just copied:

`brlpdwrappermfc9330cdw -> ../../../../opt/brother/Printers/mfc9330cdw/lpd/filtermfc9330cdw`

File: net-print/brother-mfc9330cdw-bin/brother-mfc9330cdw-bin-1.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	has_multilib_profile && ABI=x86
+
+	dosbin "${WORKDIR}/usr/bin/brprintconf_mfc9330cdw"
+
+	cp -r usr "${D}" || die
+	cp -r opt "${D}" || die
+
+
+	mkdir -p ${D}/usr/libexec/cups/filter || die
+	( cd ${D}/usr/libexec/cups/filter/ && ln -s ../../../../opt/brother/Printers/mfc9330cdw/lpd/filtermfc9330cdw brlpdwrappermfc9330cdw ) || die
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc9330cdw-bin/brother-mfc9330cdw-bin-1.1.2.ebuild]: The symlink is create here. Can you try to replace the line with the following:
( cd ${D}/usr/libexec/cups/filter/ && ln -s ../../../../opt/brother/Printers/mfc9330cdw/lpd/filtermfc9330cdw brother_lpdwrapper_mfc9330cdw ) || die

Looks like the name does not follow the same pattern as for the MFC-9320CDW but more like the MFC-9340CDW.
Which you can see here:
https://github.com/stefan-langenmaier/brother-overlay/blob/master/net-print/brother-mfc9340cdw-bin/brother-mfc9340cdw-bin-1.1.2.ebuild

**Comment by stefan-langenmaier** [comment]
Maybe you can make the modification as noted in the file and push again.

**Comment by Pistos** [comment]
Okay.  I don't really have time right now, but maybe tomorrow or the next day.  Thanks for your help!

**Comment by Pistos** [comment]
https://github.com/stefan-langenmaier/brother-overlay/pull/13

**Pull request merged**